custom/icds_reports/utils/aggregation_helpers/distributed/thr_form_v2.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from dateutil.relativedelta import relativedelta
+
+from custom.icds_reports.const import AGG_THR_V2_TABLE
+from custom.icds_reports.utils.aggregation_helpers import month_formatter
+from custom.icds_reports.utils.aggregation_helpers.monolith.base import BaseICDSAggregationHelper
+
+
+class THRFormV2AggDistributedHelper(BaseICDSAggregationHelper):
+    helper_key = 'thr-form-v2'
+    ucr_data_source_id = 'static-thr_forms_v2'
+    aggregate_parent_table = AGG_THR_V2_TABLE
+    aggregate_child_table_prefix = 'icds_db_thr_form_v2_'
+
+    def aggregate(self, cursor):
+        drop_query = self.drop_table_query()
+        curr_month_query, curr_month_params = self.create_table_query()
+        agg_query, agg_param = self.aggregate_query()
+        cursor.execute(drop_query)
+        cursor.execute(curr_month_query, curr_month_params)
+        cursor.execute(agg_query, agg_param)
+
+    def aggregate_query(self):
+        month = self.month.replace(day=1)
+        tablename = self.generate_child_tablename(month)
+        next_month_start = self.month + relativedelta(months=1)
+
+        query_params = {
+            "state_id": self.state_id,
+            "start_date": month_formatter(month),
+            "end_date": month_formatter(next_month_start)
+        }
+
+        return """
+        INSERT INTO "{tablename}" (
```

Comment:
Do I need to use the temp table here? I saw in some of the [distributed ones](https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/utils/aggregation_helpers/distributed/ls_vhnd_form.py) temp tables are created. I dont know why though.
@calellowitz  @emord

Reply:
only if the table that the data is coming from is a distributed table